# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/atulpatel-net/FlyRank_ML_In/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### Chosen Lane

I have chosen **Lane 2: Refresh / Content Opportunity Scoring**.

I selected this lane because it focuses on helping content teams identify which pages should be reviewed first based on search and content performance signals. Instead of relying on a single metric, this approach combines multiple signals to prioritize pages that may benefit from a content refresh. I also chose this lane because it provides an opportunity to apply machine learning to a real-world decision-making problem while keeping the recommendations explainable and practical.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision

The goal of this project is to improve the decision of **which content pages should be reviewed first** for a content refresh.

### Who acts on it?

The ranked recommendations are intended for **content editors or SEO teams**, who can use them to prioritize pages for review instead of checking every page manually.

### Cost of a wrong recommendation

If a page is incorrectly prioritized, time may be spent reviewing pages that do not need immediate attention, while pages that actually require updates could be missed or delayed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("content_refresh_anonymized.csv")

# Supporting statistics
total_pages = len(df)
declining_pages = (df["trend_direction"] == "down").sum()
old_pages = (df["content_age_days"] > 365).sum()

print(f"Total pages: {total_pages}")
print(f"Pages with a downward trend: {declining_pages}")
print(f"Pages older than one year: {old_pages}")

Total pages: 30000
Pages with a downward trend: 16262
Pages older than one year: 6360


### Observation

The starter dataset contains **30,000** content pages, making manual review impractical. Among them, **16,262** pages are showing a downward trend, and **6,360** pages are more than one year old. These numbers indicate that many pages could require attention, making a data-driven approach to prioritize content reviews both useful and scalable.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What my work can claim

This project can identify pages that appear to be good candidates for content review based on the available search and content signals. The recommendations are intended to support decision-making by ranking pages according to their refresh priority.

### What my work cannot claim

This project cannot prove that refreshing a page will increase its search performance or predict Google's ranking algorithm. The results should be treated as decision-support recommendations based on observed patterns in the available data, not as causal conclusions.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.